src/types.rs

```diff
@@ -848,6 +848,71 @@ impl OwnedReceiveName {
     pub fn as_ref(&self) -> ReceiveName { ReceiveName(self.0.as_str()) }
 }
 
+/// An entrypoint name (borrowed version). Expected format:
+/// "<func_name>" where
+#[derive(Eq, PartialEq, Debug, Clone, Copy, Hash)]
+pub struct EntrypointName<'a>(pub(crate) &'a str);
+
+impl<'a> EntrypointName<'a> {
+    /// Size of the name in bytes.
+    pub fn size(&self) -> u32 { self.0.as_bytes().len() as u32 }
+
+    /// Create a new name and check the format. See [is_valid_entrypoint_name]
+    /// for the expected format.
+    pub fn new(name: &'a str) -> Result<Self, NewReceiveNameError> {
+        is_valid_entrypoint_name(name)?;
+        Ok(Self(name))
+    }
+}
+
+impl<'a> From<EntrypointName<'a>> for OwnedEntrypointName {
+    fn from(epn: EntrypointName<'a>) -> Self { Self(String::from(epn.0)) }
+}
+
+/// An entrypoint name (owned version). Expected format:
+/// "<func_name>" where
+#[derive(Eq, PartialEq, Debug, Clone, Hash)]
+pub struct OwnedEntrypointName(pub String);
```

Comment:
I think this should be `pub(crate)` so you are forced to use the `new` method.

```suggestion
pub struct OwnedEntrypointName(pub(crate) String);
```

Reply:
I tend to agree, but I'll double check whether this was not needed in some ffi (otherwise we might need to add some other From/To instance or similar).

Reply:
Changed, and From implementations added.